src/plugins/index_pattern_field_editor/public/components/field_editor_flyout_content_container.tsx

```diff
@@ -53,23 +67,61 @@ export interface Props {
  * The <FieldEditorFlyoutContent /> component is the presentational component that won't know
  * anything about where a field comes from and where it should be persisted.
  */
+
 export const FieldEditorFlyoutContentContainer = ({
   field,
   onSave,
   onCancel,
   docLinks,
+  indexPatternService,
   ctx: { indexPattern, fieldTypeToProcess },
+  notifications,
+  fieldFormatEditors,
+  fieldFormats,
+  uiSettings,
 }: Props) => {
-  const fieldToEdit = deserializeField(field);
+  const fieldToEdit = deserializeField(indexPattern, field);
   const [Editor, setEditor] = useState<React.ComponentType<FieldEditorProps> | null>(null);
 
   const saveField = useCallback(
     async (updatedField: Field) => {
-      const indexPatternField = serializeField(updatedField);
-      // TODO: here we will put the logic to update the Kibana saved object
-      onSave(indexPatternField);
+      const script = updatedField.script?.source ? updatedField.script : undefined;
+
+      if (fieldTypeToProcess === 'runtime') {
+        indexPattern.addRuntimeField(updatedField.name, {
+          type: updatedField.type as RuntimeType,
+          script,
+        });
+      }
+
+      const editedField = indexPattern.getFieldByName(updatedField.name);
+
+      try {
+        if (!editedField) {
+          throw new Error(
+            `Unable to find field named '${updatedField.name}' on index pattern '${indexPattern.title}'`
+          );
+        }
+
+        indexPattern.setFieldCustomLabel(updatedField.name, updatedField.customLabel);
+        editedField.count = updatedField.popularity || 0;
+        if (updatedField.format) {
+          indexPattern.setFieldFormat(updatedField.name, updatedField.format);
+        } else {
+          indexPattern.deleteFieldFormat(updatedField.name);
+        }
+
+        indexPatternService.updateSavedObject(indexPattern).then(() => {
+          onSave(editedField);
```

Comment:
I think it would be better to put the success toast here so we have them both (success and error) in the same place.